perceval/simulators/feed_forward_simulator.py

```diff
@@ -0,0 +1,296 @@
+                    # No need for post_process here: results are already post-processed by the sub simulator
+                    res[st] = p
+
+                simulated_measures.add(measured_state)
+
+        if normalize:
+            res.normalize()
+        return res, global_perf
+
+    def _find_next_simulation_layer(self) -> tuple[list[tuple[int, AFFConfigurator]], list[int]]:
+        """
+        :return: The list containing the tuples with the index in the component list
+        of the configuration independent FFConfigurators and their instances, and the list of the associated measured modes
+        """
+        # We can add a configurator as long as the measured mode don't come from a configurable circuit
+        feed_forwarded_modes: set[int] = set()
+        measured_modes = set()
+        res = []
+
+        for i, (r, c) in enumerate(self._components):
+            if isinstance(c, AFFConfigurator):
+                if any(r0 in feed_forwarded_modes for r0 in r):
+                    return res, list(measured_modes)
+
+                feed_forwarded_modes.update(c.config_modes(r))
+                res.append((i, c))
+                measured_modes.update(r)
+
+            elif isinstance(c, Barrier):
+                continue
+
+            elif isinstance(c, PERM):
+                to_remove = []
+                to_add = []
+                perm_vector = c.perm_vector
+                for new_mode in r:
+                    if new_mode in feed_forwarded_modes:
+                        to_remove.append(new_mode)
+                        to_add.append(perm_vector[new_mode - r[0]] + r[0])
+                feed_forwarded_modes.difference_update(to_remove)
+                feed_forwarded_modes.update(to_add)
+
+            elif any(new_mode in feed_forwarded_modes for new_mode in r):
+                feed_forwarded_modes.update(r)
+
+        return res, list(measured_modes)
+
+    def _init_simulator(self, input_state,
+                        components: list[tuple[tuple, AComponent | Processor]],
+                        filter_states: bool = False,
+                        new_heralds: dict[int, int] = None):
+        """Initiate a new simulator with the given components and heralds.
+         Heralds that are already in this simulator are still considered.
+
+         :param input_state: The input state used for the simulation
+         :param components: A list of components that will be added in the simulation. Can themselves be processors.
+         :param filter_states: Whether the states should be filtered in the sub-simulation.
+         :param new_heralds: The list of heralds that should be added, containing the position and the value"""
+
+        proc = Processor(self._backend, input_state.m)
+
+        for r, c in components:
+            proc.add(r, c)
+
+        if filter_states:
+
+            if new_heralds is not None:
+                for r, v in new_heralds.items():
+                    proc.add_herald(r, v)
```

Comment:
This call adds heralds on both the input and output of `proc`. I think you want to fix heralds only in the output, which can be achieved with :

```python
proc.add_port(r, Herald(v), PortLocation.OUTPUT)
```